cpp/src/io/comp/decompression.cpp

```diff
@@ -835,11 +835,11 @@ void decompress(compression_type compression,
 {
   auto const nvcomp_type = detail::to_nvcomp_compression(compression);
   switch (compression) {
+    case compression_type::LZ4:
```

Comment:
used to return true w/o nvcomp, which is incorrect. The error was not caught because until now we always used to check both decompression and compression.